### ISSUE TITLE
X and Y higher resolution and bed temp and nozzle temp upvote

### ISSUE THREAD

**Issue by Omnipresent77** [j-be/AutoBim#38]
I am sorry that I am unfamiliar enough to know how a pull request works. 

I would like to request that when setting the four test positions, can two decimal places beyond the whole number be used also?
 like instead of x:30 y:30  it could be x29.85 y:30.05?

Also I would like to second the notion of what i seen request in another of being able in a input, or drop down if you can figure out how to heat up the bed and nozzle both.  I know when I do the mesh, ( the bi-linear one?) to really zero in the bed i need to heat up both the bed and nozzle.

I am on a Ender 3 pro. BLtouch, hemera all-in-one head, BRR turbo board.

But yes I would like to officially upvote that possibility of adding a function to preheat bead and nozzle to a temp first .

..and also adding a bit more resolution to the X and Y points locations.

**Comment by j-be** [comment]
Hi, no worries. The important thing is you found a way to reach out :smile: 

I think more decimal places should work out-of-the-box, up until the limit your firmware accepts (which is usually a lot more than 2). So putting in `29.85` for X and `30.05` should work just fine.

Also, you can already pre-heat bed and nozzle. Use the "Before" in "Custom GCode" in the plugin's settings and add something like:

```
M190 S60
```

This would heat up the bed to 60 degree before starting the first round.

**However**: The plugin is not designed for waiting too long for something to happen. In more detail this means: If the plugin doesn't get a valid response from the printer within 3 minutes it will assume something went wrong and abort the run. So keep an eye on the OctoPrint UI while trying this and if you run into timeouts while preheating, let me know. I think it shouldn't be too hard to implement something to let you get around that.

Oh, final remark: you really shouldn't need to heat up your nozzle. The nozzle temp should not influence the bed geometry, which is what we are dealing with here. It may affect Z offset, but that is not in scope here. Actually, quite the opposite: a heat source near the BLTouch while it is measuring may actually do more harm than good.

If you really must heat nozzle, too, (I again strongly advise against it) you can use the following GCode in "Before":

```
M140 S60 ; Start heating bed to 60 degrees, but don't wait
M104 S200 ; Start heating nozzle, but don't wait
M190 S60 ; Wait for bed to reach 60 degrees
M109 S200 ; Wait for nozzle to reach 200 degrees
```

For reference:

* https://marlinfw.org/docs/gcode/M140.html
* https://marlinfw.org/docs/gcode/M104.html
* https://marlinfw.org/docs/gcode/M190.html
* https://marlinfw.org/docs/gcode/M109.html
